src/ucp/core/ucp_context.c

```diff
@@ -1462,6 +1466,14 @@ static ucs_status_t ucp_fill_config(ucp_context_h context,
         goto err;
     }
 
+    /* save comparison MD for iface_attr adjustment */
+    context->config.ext.selection_cmp = ucs_strdup(config->ctx.selection_cmp,
+                                                   "selection cmp");
+    if (context->config.ext.selection_cmp == NULL) {
+        status = UCS_ERR_NO_MEMORY;
+        goto err;
```

Comment:
if this failed, we don't release env_prefix

Reply:
fixed